LSPS0/README.md

```diff
@@ -0,0 +1,606 @@
+entropy for universal uniqueness.
+
+Clients SHOULD internally impose a reasonable timeout, on
+the scale of minutes, for receiving a response for a request,
+and SHOULD treat a timeout event as a temporary server failure,
+and forget the `id` of the timed-out request.
+
+> **Rationale** The LSP might crash between the time the client
+> makes the request to the time it could complete processing of
+> the response, and thus lost track of the request.
+
+If the LSP is unable to deliver a response to the client due
+to a disconnection, it SHOULD treat this no differently from
+successfully delivering the response but the client then does
+not "follow up" on any action after that.
+
+> **Rationale** Even if the LSP delivers the response to the
+> client, the client could then crash while processing the
+> response, which means that the LSP cannot be sure that any
+> response is received by the client anyway.
+
+### Lightning Feature Bit
+
+The [BOLT7][] specification describes the `node_announcement`
+gossip message, which includes a `features` bit field.
+The [BOLT1][] specification describes the `init` message,
+which also includes a `features` bit field.
+
+[BOLT7]: https://github.com/lightning/bolts/blob/f7dcc32694b8cd4f3a1768b904f58cb177168f29/07-routing-gossip.md
+[BOLT1]: https://github.com/lightning/bolts/blob/f7dcc32694b8cd4f3a1768b904f58cb177168f29/01-messaging.md
+
+LSPs MUST set the `features` bit numbered 729
+(`option_supports_lsps`) in both the `init` message on connection
+establishment, and in their own advertised `node_announcement`.
+Clients MUST NOT set `features` bit numbered 729 in either
+context.
+
+> **Rationale** As clients are the one who must initiate any
+> BOLT8 message ID 37913 messages before LSPs respond, they need to
+> separately discover whether a peer they connect to is
+> indeed an LSPS LSP that recognizes that message.
+> Broadcasting this as part of gossip allows a client to
+> discover new LSPs by simply downloading gossip.
+> The bit 729 was chosen randomly and has no special meaning.
+
+### LSPS Specification Support Query
+
+The client can determine if an LSP supports a particular LSPS
+specification other than LSPS0 via the `method` named
+`lsps0.listprotocols`, which has the following parameters:
+
+```JSON
+{
+  "comment": "This comment is ignored"
+}
+```
+
+`comment` is an optional string, and is ignored by the LSP.
+It defaults to the empty string.
+
+> **Rationale** In addition to allowing clients to determine what
+> protocols an LSP supports, this query is also an example of what
+> other LSPS specifications need to describe, such as the parameters
+> of an API endpoint, the type of each parameter, whether each
+> parameter is optional, the default value if the parameter is
+> optional but not specified by the client, and what each parameter
+> means for the LSP.
+
+`lsps0.listprotocols` has no errors defined.
+
+The response datum is an object like the below:
+
+```JSON
+{
+  "protocols": [1, 3]
```

Comment:
Does it make sense to turn these into strings, so we can add protocol versions (1.1, 3.1) later?

Reply:
Intent is that protocol versions are handled by the specific LSPS specification itself, e.g. `versions` array from `GET /jitchannels` (which will have to be renamed to be LSPS0-compliant, once we have a stable LSPS0).